### PR TITLE
got: 0.97 -> 0.98.2

### PR DESCRIPTION
**Pull request by afh**
## Description of changes

Upstream has switched to libressl (see [`ThomasAdam/got-portable@62e037f`](https://github.com/ThomasAdam/got-portable/commit/62e037f1de942562a70addbbb4695c524f660160))

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by JohnRTitor** [comment]
Note: PR title should be updated to properly indicate this is a `got` update, not `git`.

**Comment by afh** [comment]
Great catch, @JohnRTitor! The muscle memory in my fingers got (pun intended) the better of me ;)

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 307538` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>got</li>
  </ul>
</details>

**Review by JohnRTitor — APPROVED** [review]
LGTM.

**Review by AndersonTorres — APPROVED** [review]

**Comment by JohnRTitor** [comment]
@NixOS/nixpkgs-merge-bot merge

**Comment by AndersonTorres** [comment]
@afh can you migrate this to by-name?

**Comment by JohnRTitor** [comment]
Is there a "general" guideline on which packages should be under a category or in `by-name`?

File: pkgs/by-name/go/got/package.nix [hunk 1]
```diff
@@ -45,13 +46,9 @@ stdenv.mkDerivation (finalAttrs: {
...
-  doInstallCheck = true;
-
-  installCheckPhase = ''
-    runHook preInstallCheck
-    test "$($out/bin/got --version)" = "got ${finalAttrs.version}"
-    runHook postInstallCheck
-  '';
+  passthru.tests.version = testers.testVersion {
+    package = finalAttrs.finalPackage;
+  };
```
Comment (afh) [inline pkgs/by-name/go/got/package.nix:51]: @AndersonTorres, is this change a proper and correct improvement? I have yet to fully understand `passthru` and `testers.testVersion`, but from my current understanding this should be good. I'd truly appreciate your feedback and expertise on this.

**Comment by afh** [comment]
Sure thing, @AndersonTorres 👍 Please see comment above for a question I have about an additional change I made regarding `installCheck`.

**Comment by AndersonTorres** [comment]
> Is there a "general" guideline on which packages should be under a category or in `by-name`?

A very rough guideline is: `by-name` for the win!

- New packages should be on `by-name` except if they can't
  - Example: there are packages that use `darwin.apple_sdk.VERSION.callPackage`; they can't use by-name (yet), so use the old hierarchy for them
- Old packages can, optionally, be migrated to `by-name` (except if they can't - see above) **only** as part of refactoring or version update
  - This is because a mass migration is being planned; so, do not migrate an old package except in the cases above

File: pkgs/by-name/go/got/package.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/by-name/go/got/package.nix:51]: Yes, it is fine.

As I have explained before, `installCheckPhase` is meant to run checks provided by the upstream package developer. The difference between `checkPhase` and `installCheckPhase` is that `installCheckPhase` runs after installation while `checkPhase` runs after build.

Custom scripts are better suited to `passthru.tests`.

Comment (afh) [reply pkgs/by-name/go/got/package.nix:51]: Thanks for the explanation, @AndersonTorres, that's helpful!

Yet for some odd reason my 🧠 refuses to understand `passthru` (might be something to do with the wording that seems odd to me—not that it is, just that it makes it harder for me to grok it). Seems I need to revisit and re-read the [available documentation](https://ryantm.github.io/nixpkgs/stdenv/stdenv/#var-stdenv-passthru)…

Comment (AndersonTorres) [reply pkgs/by-name/go/got/package.nix:51]: `passthru` is similar to `meta` in the sense that a modification in `passthru` does not trigger a mass rebuild.

Usually we use `passthru` in order to collect some useful info from the derivation and expose it to the external world.

E.G. suppose you have a parameter, `uiType`, that enables a fancy UI in `got`, and the end user wants to know which GUI type was used (say, `cli, curses, gtk3, qt10`).

Then you can write

```nix
passthru = {
   uiType =
     if enableCurses then "curses"
     else if enableGtk3 then "gtk3"
     else "cli";
};
```

This way, the end user can evaluate `got.uiType` to know what UI was enabled.

------------------------------

Obviously the code above is a crude example. Usually we do cleaner and more useful stuff.

E.G. in `live555` we put a test: 

```nix
passthru.tests = {
   inherit vlc;
};
```

This way we can test this by calling `live555.tests.vlc`.

The idea is that `vlc` depends on `live555`, and sometimes the `vlc` team does not update its code when `live555` introduces a breaking change. So we at Nixpkgs catch this by running this test.

**Comment by afh** [comment]
Thanks for the review, @AndersonTorres, and thanks for merging, @Mic92, much appreciated 🙂👍

**Pull request merged**